control/init.sh

```diff
@@ -17,14 +18,21 @@ function realpath() {
 
 
 if [ -z ${CI_TEST} ]; then
-    if [ ! -f $VENV/bin/activate ]; then
-        if [[ $CCHQ_VIRTUALENV == *3.10* ]]; then
-          # use venv because 3.10 setup includes installing python3.10-venv
-          python3.10 -m venv $VENV
+    if [[ $USE_SYSTEM_PYTHON == true ]] && hash python3.10 2>/dev/null && [[ $( source /etc/os-release; echo $VERSION_ID ) == 18.04 ]]; then
+        # if on 18.04 with 3.10 installed, use cchq-3.10 unless $USE_SYSTEM_PYTHON is true
+        CCHQ_VIRTUALENV=$CCHQ_VIRTUALENV-3.10
+        VENV=~/.virtualenvs/$CCHQ_VIRTUALENV
+    fi
+    # check if a virtualenv at $VENV exists yet, and create if not
+    if [[ ! -f $VENV/bin/activate ]]; then
+        if hash python3.10 2>/dev/null; then
```

Comment:
This will still use Python 3.10, even if `$USE_SYSTEM_PYTHON == true`, and will create a 3.10 env named `cchq` on Ubuntu 18.04 if python 3.10 is installed and a 3.6 virtualenv with that name does not exist. I think that would be a confusing state to get into.

Reply:
See d7b796ffecf8df3ef2b5635456424a572a1c79fe